src/toxcore/dht/server/mod.rs

```diff
@@ -691,10 +720,30 @@ impl Server {
         };
 
         let close_nodes = close_nodes.get_closest(&self.pk, IsGlobal::is_global(&addr.ip()));
+
+        let mut collected_bucket = Bucket::new(Some(4));
+
+        close_nodes.iter()
+            .for_each(|node| {
+                collected_bucket.try_add(&self.pk, node);
```

Comment:
Why `self.pk` is used? Shouldn't it be payload.pk? And why do we need to iterate friends here?

Reply:
payload.pk should be used.
When we respond with NodesResponse, we should search not only close_list but also each friend's close_list. It was omitted in previous PR.